docs/designs/upgrade-design/upgrade-design-doc.md

```diff
@@ -0,0 +1,653 @@
+
+### Phase 2: Cluster Migration Coordination (Database Admin)  
+**Responsibility**: Plan and coordinate DocumentDB cluster API version migrations
+**Scope**: Per-cluster API version upgrades across environments
+**Method**: Coordinate with Database Developers for phased migration execution
+
+**What gets planned and coordinated**:
+- Cluster API migration timeline across dev/test/staging/production
+- Migration dependencies and environment-specific requirements
+- Database configuration updates for v2 API features
+- Cluster-level backup and validation procedures
+- Performance and capacity impact assessment
+
+**Database Admin Control**:
+- Plans migration schedule across all environments
+- Coordinates with Database Developers for execution
+- Manages cluster configurations and database-level settings
+- Oversees backup and recovery procedures during migration
+### Phase 3: Application Migration Execution (Database Developer)
+**Responsibility**: Execute cluster API migration and validate application compatibility
+**Scope**: Application-specific testing and validation
+**Method**: Update API version in deployment files and execute migration with comprehensive testing
+
+**What gets executed**:
+- Specific DocumentDB cluster API (v1 → v2) migration
+- Application compatibility testing with v2 API features
+- Client application integration validation
+- Data integrity verification during migration
+- Performance testing and optimization
+
+**Database Developer Control**:
+- Executes migration based on Database Admin's coordination plan
+- Gradual rollout across development, staging, production environments
+- Application-specific testing and validation with v2 features
+- Rollback capability per cluster (v2 → v1 downgrade) if application issues occur
+- Test coverage across environments to ensure application compatibility
+
+**Benefits of Three-Tier Responsibility Model**:
+- **Clear Role Separation**: Each team focuses on their domain expertise
+- **Coordinated Migration**: Database Admins coordinate while Database Developers execute
+- **Gradual API Migration**: Database Developers control application-specific migration timeline
+- **Risk Mitigation**: Test new API versions in dev/staging before production migration
+- **Backward Compatibility**: Multiple cluster API versions supported simultaneously
+- **Controlled Deprecation**: API versions deprecated gradually over multiple operator releases
+
+**Cross-Team Coordination**: Database Admins coordinate migration planning with Database Developers for execution, while Kubernetes Admins provide the infrastructure foundation.
+
+**API Version Examples**: See [commands.md](./commands.md) for detailed workflow commands and API version migration examples.
+
+## Component-Specific Upgrade Considerations
+
+While all components upgrade together, each has specific characteristics:
+
+#### Gateway Image Upgrade (API Version Dependent)
```

Comment:
Gateway should be updated lockstep with database, e.g. we should *not* see gw v2, postgres v1

Reply:
is this not CNPG managed as well?

Reply:
Yes. all the component versions are locked with our operator version. customer cannot choose individual component versions.

for the second part, our gateway image is managed by the CNPG sidecar injector only. Whatever image:tag we give for gateway, injector just use it.